File: p004/main.py
#!/usr/bin/env python3

def is_palin(n: int) -> bool:
    s = str(n)
    reversed = s[::-1]
    return s == reversed

def run(start: int, stop: int) -> None:
    found: list[int] = []
    for x in range(start, stop):
        for y in range(start, stop):
            n = x * y
            if is_palin(n):
                found.append(n)
    print("Largest palindrome:", max(found))

run(100, 999)

File: p004/main.rs
fn is_palin(s: &str) -> bool {
    let r = s.chars().rev().collect::<String>();
    return s == r;
}

fn main() {
    let mut found: Vec<i32> = vec![];
    for x in 100..=999 {
        for y in 100..=999 {
            let n = x * y;
            let s = n.to_string();
            if is_palin(&s) {
                found.push(n);
            }
        }
    }
    let max = found.iter().max();
    match max {
        Some(max) => println!("Largest palindrome: {}", max),
        None => println!("No palindromes found"),
    }
}
